CommonTools/UtilAlgos/interface/ObjectSelectorProducer.h

```diff
@@ -56,6 +56,12 @@ class ObjectSelectorProducer : public Base {
     selectorInit_.init(selector_, evt, es);
     edm::Handle<typename Selector::collection> source;
     evt.getByToken(srcToken_, source);
+    if (!source.isValid()) {
+      // optionally log something
+      edm::LogInfo("ObjectSelectorProducer") << "Source collection not found for token. Skipping event.";
```

Comment:
can this silent skip be made configurable and off by default?
I'm thinking of RECO applications where (at least from my days) the expectation was that all products are available and if not it's a configuration error

Reply:
I was thinking the same, applied in the last push.